### PR TITLE
fix: update lint config, fix lint errors, move lint to its own github action

### PR DESCRIPTION
**Pull request by noranda**
## What did we change?

This PR does the following:
- Bumps `@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser` from `4.16.1` to `5.15.0`
- Adds `eslint-config-prettier` and `eslint-plugin-prettier`
- Updates `.eslintrc.js` config:
  - Fixes issue with parser which was nested incorrectly causing erroneous typescript errors
  - Adds `plugin:prettier/recommended` to extensions to resolve conflicts between prettier and eslint rules
  - Removes rules that are no longer needed/used
  - Specifies `devDependencies` to exclude from the `import/no-extraneous-dependencies` rule
  - Adds `overrides` files with specific rules allowing linting for previously ignored files (like tests)
- Fixes all linting errors
- Moves linting out of the `test.yml` github workflow and into its own `code_quality.yml` workflow

## Why are we doing this?

Linting is an important way to keep a code clean, consistent, and maintainable. Due to issues with the test github workflow, which linting was part of, linting hasn't been up to date and various lint errors have appeared. This PR brings linting back up to code, fixing errors, updating the config file, and moving linting to its own workflow so it can continue to keep the code base clean. 

This only affects the internal working of Recipe.

## Checklist

- [x] Create a changeset (`npm run changeset`) with a summary of the changes

### CODE REVIEW

**Review by pgrippi — DISMISSED** [review]
The [new workflow looks good](https://github.com/ezcater/recipe/runs/5630146295?check_suite_focus=true)!

> 0 error(s), 0 warning(s) found

**Review by pgrippi — APPROVED** [review]

File: .changeset/spotty-ghosts-hug.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@ezcater/recipe': patch
+---
+
+fix: updated lint config, fixed lint errors, moved lint to its own github action
```
Comment (danidewitt) [inline .changeset/spotty-ghosts-hug.md:5]: 🤷‍♀️ could also be `workflow:` (not blocking)

File: .github/workflows/code_quality.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+name: Code Quality
```
Comment (danidewitt) [inline .github/workflows/code_quality.yml:1]: 👏

File: src/components/EzSegmentedControl/EzSegmentedControl.tsx [hunk 3]
```diff
@@ -17,7 +17,7 @@ type Button = {
 type Props = Omit<React.FieldsetHTMLAttributes<any>, 'onChange'> & {
   active?: any;
   label: React.ReactNode;
-  labelPosition: 'left' | 'hidden';
+  labelPosition?: 'left' | 'hidden';
```
Comment (danidewitt) [inline src/components/EzSegmentedControl/EzSegmentedControl.tsx:20]: ❓ was this optional before?

File: .changeset/spotty-ghosts-hug.md (hunk 1 above)
Comment (danidewitt) [reply .changeset/spotty-ghosts-hug.md:5]: ok now that i looked at _everything_ else, disregard my previous comment

**Review by danidewitt — APPROVED** [review]

File: src/components/EzSegmentedControl/EzSegmentedControl.tsx (hunk 3 above)
Comment (noranda) [reply src/components/EzSegmentedControl/EzSegmentedControl.tsx:20]: @danidewitt It has a default prop listed, so it looks as if it was meant to be optional.

**Review by pgrippi — APPROVED** [review]

**Review by danidewitt — APPROVED** [review]

**Pull request merged**